frontend/app.js

```diff
@@ -0,0 +1,510 @@
+var React = require('react');
```

Comment:
You should try to write some tests for your javascript as well, but I wouldn't stress too much about testing your components. Component testing doesn't add that much value, IMO.

Instead, I would try to extract the business logic (e.g. AJAX requests and state updates) from your components and test that stuff. The goal should be to have your components be as dumb as possible and just worry about rendering.

Because your business logic is kind of coupled to your components (because you're relying on component state), this is kind of a challenge. This is probably the biggest reason that people adopt redux for larger projects. This project definitely doesn't *need* redux, but you should learn it, so that when you do work on a real project, you know it.